packages/snack-sdk/src/DependencyResolver.ts

```diff
@@ -112,7 +112,7 @@ export default class DependencyResolver {
         fileName: key,
       });
       this.logger?.error(error);
-      if (this.status[key]) {
+      if (key) {
```

Comment:
This seems like a behavioral change, can you give some context on why this was necessary?

Reply:
`this.status[key]` always returns a Promise, which causes a type error. As all that is being done in that if statement is deleting the relevant item for the key and, since the key is of type `string`, you can simply check for the key instead of checking for the item for said key.

Reply:
if left as a Promise, there will both be a type error and it will always return `true`

Reply:
I'm not sure if I follow, the `this.status` dictionary should contain the (latest?) promises related to dependency actions. So I guess it's an important part of checking if something is going on right?

We have this check in multiple places, e.g. https://github.com/expo/snack/blob/main/packages/snack-sdk/src/DependencyResolver.ts#L76-L86. And if there is no pending action going on, then it shouldn't try to delete it right?

I agree on the "all that is being done is deleting the relevant status", but I'm not sure if that always will be like that. Plus it does log something, which means it could trigger a log while that log it isn't relevant, right? That log might confuse users maybe?